src/main/java/xyz/aadev/generitech/client/gui/upgrade/GuiUpgradeScreen.java

```diff
@@ -0,0 +1,146 @@
+ */
+
+import net.minecraft.client.gui.GuiButton;
+import net.minecraft.client.renderer.RenderHelper;
+import net.minecraft.client.renderer.RenderItem;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.InventoryPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.tileentity.TileEntity;
+import xyz.aadev.aalib.client.gui.GuiBase;
+import xyz.aadev.aalib.common.util.GuiHelper;
+import xyz.aadev.generitech.Reference;
+import xyz.aadev.generitech.api.util.MachineTier;
+import xyz.aadev.generitech.client.gui.button.ButtonSides;
+import xyz.aadev.generitech.common.blocks.Blocks;
+import xyz.aadev.generitech.common.container.upgrade.ContanierUpgradeStorage;
+import xyz.aadev.generitech.common.network.Network;
+import xyz.aadev.generitech.common.network.messages.power.PacketSides;
+import xyz.aadev.generitech.common.tileentities.TileEntityMachineBase;
+
+import java.awt.*;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class GuiUpgradeScreen extends GuiBase {
+
+    TileEntity tileEntity;
+    GuiHelper guiHelper = new GuiHelper();
+    EntityPlayer player;
+    Rectangle slot;
+    private int[] sides;
+    private int machineTier;
+
+
+    public GuiUpgradeScreen(InventoryPlayer inventoryPlayer, TileEntity tileEntity, int[] sides, int start, EntityPlayer player) {
+        super(Reference.MOD_ID, new ContanierUpgradeStorage(inventoryPlayer, tileEntity, start));
+        this.sides = sides;
+        this.xSize = 176;
+        this.ySize = 166;
+        this.player = player;
+        this.tileEntity = tileEntity;
+
+        machineTier = MachineTier.byMeta(tileEntity.getBlockMetadata()).getTier();
+        slot = new Rectangle(48, 31, 25, 24);
+
+    }
+
+    @Override
+    public void drawBG(int paramInt1, int paramInt2, int paramInt3, int paramInt4) {
+        bindTexture("gui/upgrade/upgrade.png");
+        drawTexturedModalRect(paramInt1, paramInt2, 0, 0, this.xSize, this.ySize);
+        if (tileEntity instanceof TileEntityMachineBase) {
+            sides = ((TileEntityMachineBase) tileEntity).getSides();
+        }
+    }
+
+    @Override
+    public void drawFG(int paramInt1, int paramInt2, int paramInt3, int paramInt4) {
+        // Do nothing because of not implemented
+    }
+
+
+    @Override
+    public void initGui() {
+        if (tileEntity instanceof TileEntityMachineBase) {
+            this.mc.thePlayer.openContainer = this.inventorySlots;
+            this.guiLeft = (this.width - this.xSize) / 2;
+            this.guiTop = (this.height - this.ySize) / 2;
+            int Forward = ((TileEntityMachineBase) tileEntity).getForward().getIndex();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS00117)